pkg/cmd/cli/cmd/rsync.go

```diff
@@ -0,0 +1,372 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/tar"
+	"github.com/spf13/cobra"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kubecmd "k8s.io/kubernetes/pkg/kubectl/cmd"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	// RsyncRecommendedName is the recommended name for the rsync command
+	RsyncRecommendedName = "rsync"
+
+	rsyncLong = `
+Copy local files to a container
+
+This command will attempt to copy local files to a remote container. It will default to the
+first container if none is specified, and will attempt to use 'rsync' if available locally and in the
+container. If 'rsync' is not present, it will attempt to use 'tar' to send files to the container.`
+
+	rsyncExample = `
+  # Synchronize a local directory with a pod directory
+  $ %[1]s POD ./local/dir/ /remote/dir`
```

Comment:
So 'oc rsync ./local/dir [POD]:/remote/dir' ? I like that better

Hmm, and yes, I suppose we could also allow 
'oc rsync [POD]:/remote/dir ./local/dir', 
My understanding is that initially we only wanted client -> pod though.